dags/analyzer_helper/tests/unit/test_transform_raw_info.py

```diff
@@ -0,0 +1,370 @@
+                    'interactions': [
+                        {
+                            'name': 'reaction',
+                            'type': 'emitter',
+                            'users_engaged_id': ['6261'],
+                        }
+                    ],
+                    'metadata': {
+                        'category_id': None,
+                        'topic_id': 6134,
+                        'bot_activity': False,
+                    },
+                    'source_id': '6261',
+                }
+            ]
+
+            result = self.transformer.transform(
+                raw_data=raw_data,
+            )
+            print("Result: \n", result)
+            print("Expected result: \n", expected_result)
+            self.assertEqual(result, expected_result)
+
+    def test_transform_data_replied_and_reactions(self):
+        raw_data = [
+            {
+                "post_id": 6262,
+                "author_id": 6168,
+                "created_at": "2023-09-11T21:41:43.553Z",
+                "author_name": "Test Author Name1",
+                "reactions": [],
+                "replied_post_id": 6512,
+                "replied_post_user_id": 4444,
+                "topic_id": 6134
+            },
+            {
+                "post_id": 6261,
+                "author_id": 6168,
+                "created_at": "2023-09-11T21:42:43.553Z",
+                "author_name": "Test Author Name2",
+                "reactions": [1, 2],
+                "replied_post_id": None,
+                "replied_post_user_id": None,
+                "topic_id": 6134
+            }
+        ]
+         
+        expected_result = [
+            {
+                'author_id': '6168',
+                'date': datetime.datetime(2023, 9, 11, 21, 41, 43, 553000, tzinfo=datetime.timezone.utc),
+                'source_id': '6262',
+                'metadata': {
+                    'category_id': None,
+                    'topic_id': 6134,
+                    'bot_activity': False
+                },
+                'actions': [
+                    {
+                        'name': 'message',
+                        'type': 'emitter'
+                    }
+                ],
+                'interactions': [
+                    {
+                        'name': 'reply',
+                        'type': 'emitter',
+                        'users_engaged_id': ['4444']
+                    }
+                ]
+            },
+            {
+                'author_id': '4444',
+                'date': datetime.datetime(2023, 9, 11, 21, 41, 43, 553000, tzinfo=datetime.timezone.utc),
+                'source_id': '6262',
+                'metadata': {
+                    'category_id': None,
+                    'topic_id': 6134,
+                    'bot_activity': False
+                },
+                'actions': [],
+                'interactions': [
```

Comment:
actions field should be filled if there's a reply as it contains a message too.

Reply:
Good catch. yes, we need the action field in line 120 of it should be filled too.